src/System.Memory/tests/ReadOnlySpan/GcReporting.cs

```diff
@@ -0,0 +1,67 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.SpanTests
+{
+    public static class ReadOnlySpanGcReportingTests
+    {
+        [Fact]
+        public static void DelegateTest()
+        {
+            DelegateTest(100000, 10000);
+        }
+
+        public static void DelegateTest_Stress()
+        {
+            DelegateTest(100, 100);
+        }
+
+        private static void DelegateTest(int iterationCount, int objectCount)
+        {
+            object[] objects = new object[objectCount];
+            Random rng = new Random();
+            var delegateTestCore =
+                new DelegateTestCoreDelegate(DelegateTest_Core) +
+                new DelegateTestCoreDelegate(DelegateTest_Core);
+
+            for (int i = 0; i < iterationCount; i++)
+            {
+                DelegateTest_CreateSomeObjects(objects, rng);
+
+                var spanContainer = new SpanContainer();
+                spanContainer._span = new ReadOnlySpan<int>(new int[] { 1, 2, 3 });
+                delegateTestCore(spanContainer, objects, rng);
+            }
+        }
+
+        private delegate void DelegateTestCoreDelegate(SpanContainer spanContainer, object[] objects, Random rng);
+
+        private static void DelegateTest_Core(SpanContainer spanContainer, object[] objects, Random rng)
+        {
```

Comment:
You can cache the incoming span in a local here, and do Assert at the end that the cached local copy is still the same as the incoming one.